File: examples/java-repo/src/main/java/com/example/ecommerce/domain/product/Product.java
package com.example.ecommerce.domain.product;

import com.example.ecommerce.domain.order.Money;

public class Product {

    private final long id;
    private final String name;
    private final String description;
    private final Money price;

    public Product(long id, String name, String description, Money price) {
        if (id <= 0) {
            throw new IllegalArgumentException("product id must be positive");
        }
        if (name == null || name.isBlank()) {
            throw new IllegalArgumentException("product name must not be blank");
        }
        if (price == null) {
            throw new IllegalArgumentException("price must not be null");
        }
        this.id = id;
        this.name = name;
        this.description = description != null ? description : "";
        this.price = price;
    }

    public long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getDescription() {
        return description;
    }

    public Money getPrice() {
        return price;
    }
}

File: examples/python-repo/ecommerce/domain/product/product.py
class Product:
    def __init__(self, product_id, name, description, price):
        if product_id <= 0:
            raise ValueError("product id must be positive")
        if not name or not name.strip():
            raise ValueError("product name must not be blank")
        if price is None:
            raise ValueError("price must not be None")

        self._id = product_id
        self._name = name
        self._description = description or ""
        self._price = price

    @property
    def id(self):
        return self._id

    @property
    def name(self):
        return self._name

    @property
    def description(self):
        return self._description

    @property
    def price(self):
        return self._price
